### PR TITLE
Customisation

### PR DESCRIPTION
**Pull request by valeriadamante**

### CODE REVIEW

File: Corrections.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def Initialize(config, load_corr_lib=True, load_pu=True, load_tau=True, load_trg
     global met
     global trg
     global btag
+    global sf_to_apply
```
[the remark this answers is not in the document]
Comment (valeriadamante) [reply Corrections.py:32]: ok

File: tau.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ def __init__(self, period, config):
             tauType_map = createTauSFTypeMap(config["genuineTau_SFtype"])
             ROOT.gInterpreter.ProcessLine(f'::correction::TauCorrProvider::Initialize("{jsonFile}", "{self.deepTauVersion}", {wp_map_cpp}, {tauType_map})')
             TauCorrProducer.initialized = True
+            deepTauVersion = f"""DeepTau2017{config["deepTauVersion"]}"""
```
[the remark this answers is not in the document]
Comment (valeriadamante) [reply tau.py]: ok

**Review by kandrosov — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
-__pycache__
\ No newline at end of file
+__pycache__
+data/*/*/*.root
```
[the remark this answers is not in the document]
Comment (valeriadamante) [reply .gitignore:2]: ok

**Review by kandrosov — APPROVED** [review]

File: Corrections.py [hunk 4]
```diff
@@ -132,7 +136,7 @@ def getNormalisationCorrections(df, config, sample, ana_cache=None, return_varia
     df = df.Define('genWeightD', 'std::copysign<double>(1., genWeight)')
     scale = 'Central'
     df, tau_SF_branches = tau.getSF(df, config)
-    all_branches = [ pu_SF_branches, tau_SF_branches ]
+    all_branches = [ pu_SF_branches ]
```
Comment (kandrosov) [inline Corrections.py:139]: this is correct, but we want to have product of all tauID SF as a separate branch. Better to implement it here and return them inside `all_weights`

**Review by kandrosov — CHANGES_REQUESTED** [review]

File: Corrections.py [hunk 5]
```diff
@@ -140,7 +140,19 @@ def getNormalisationCorrections(df, config, sample, ana_cache=None, return_varia
     all_sources = set(itertools.chain.from_iterable(all_branches))
     all_sources.remove(central)
     all_weights = []
+    tauID_weights = []
     denom = f'/{ana_cache["denominator"][central][central]}' if ana_cache is not None else ''
+    for scale in ["Up", "Down"]:
```
Comment (kandrosov) [inline Corrections.py:145]: - doesn't work for central use the same loop structure as for the other weights
- don't need to store for each tau separately. One branch for TauID sf is fine

**Review by kandrosov — CHANGES_REQUESTED** [review]

File: Corrections.py (hunk 5 above)
Comment (valeriadamante) [reply Corrections.py:145]: ok

File: Corrections.py [hunk 6]
```diff
@@ -132,11 +136,32 @@ def getNormalisationCorrections(df, config, sample, ana_cache=None, return_varia
     df = df.Define('genWeightD', 'std::copysign<double>(1., genWeight)')
     scale = 'Central'
     df, tau_SF_branches = tau.getSF(df, config)
```
Comment (kandrosov) [inline Corrections.py:138]: run `tau.getSF` and corresponding loop only `if 'tauID' in sf_to_apply`

**Review by kandrosov — CHANGES_REQUESTED** [review]
remove the old btagEff_TT.root and use that have more stat to compute btag SFs

File: Corrections.py (hunk 6 above)
Comment (valeriadamante) [reply Corrections.py:138]: ok

**Review by kandrosov — APPROVED** [review]

**Pull request merged**